site/docs/v1/tech/migration-guide/general.adoc

```diff
@@ -661,6 +663,65 @@ This approach has some downsides:
 
 But if you are trying to migrate an expected mass of new users and want to use FusionAuth for the new users, but are comfortable running the legacy system for older users, this approach may work.
 
+== Special Considerations
+
+Beyond migrating users, there may be other important data to import from a previous user identity datastore.
+
+=== Migrating Refresh Tokens
+
+If you have a token based authentication system, you may follow the common practice of using short-lived access tokens to protect resources, but long-lived refresh tokens to avoid needlessly reauthenticating users. If updating this system to use FusionAuth as a token source, you can avoid invalidating all the existing refresh tokens.
+
+You can import users' refresh tokens from the existing system into FusionAuth. Importing refresh tokens ensures that your users can enjoy an uninterrupted application experience. The next time their access token expires, each client can present a valid refresh token to FusionAuth, which will in turn issue a new signed access token.
+
+This option is only applicable for big-bang migrations, since a slow migration can re-issue a refresh token for each user at login time.
+
+Consult the link:/docs/v1/tech/apis/users#import-refresh-tokens[documentation for the Refresh Token Import API] for more information on how to import these tokens.
+
+=== Migrating SAML Configuration
+
+If you have SAML identity providers (IdPs) to which your existing user account system delegates, you must update all external IdP configurations that depend on an existing authentication system. This is common when you have multiple organizations represented in your application and each organization has one or more SAML identity providers which are the system of record for the organization's users in your application.
+
+When migrating, there is an existing SAML Service Provider (SP). This could be a custom application or another vendor. After migration, you will use FusionAuth as your new SP.
+
+There are two options to update this type of SAML configuration external to FusionAuth. 
+
+==== Updating Upstream Configuration
+
+The first, recommended, option, is to contact every upstream SAML IdP and have the administrators update their configuration to point to FusionAuth instead of the existing SAML SP. You can preload the required certificates in FusionAuth using the link:/docs/v1/tech/core-concepts/key-master[Key Master UI] or APIs. You can also configure link:/docs/v1/tech/identity-providers/samlv2/[FusionAuth SAMLv2 Identity Providers] with the same values (audience, etc) as the existing SAML SP.
+
+However, in this option, each upstream IdP needs to update their Assertion Consumer Service (ACS) URL to a valid FusionAuth value. Since the ACS URL typically is different and the administrators of the IdP may not make such changes a priority, this updating process may take a while.
```

Comment:
I would move this and line 694 down under the "Relaxing" section. It sounds like we are discouraging the "right way" here, which I don't think we want to do.

Reply:
Fair.